### DISCUSSION TITLE
2.5 gpio not working

### DISCUSSION THREAD

**Discussion by derschelli** [MiczFlor/RPi-Jukebox-RFID#2055] [General]
Hello there,

i'm setting up a new phoniebox. unfortunately gpio-buttons don't seem to work.

i used the developer-version, because i wanted to use the sync-feature.

after doing a fresh install, i tried to set up the hardware button. for testing i just tried one button.

connected it to pin 29 (gpio5) and 30 (ground),

changed

~/RPi-Jukebox-RFID/settings/gpio_settings.ini

to

[PlayPause]
enabled: True
Type: Button
Pin: 29
pull_up_down: pull_up
functionCall: functionCallPlayerPause

and restarted the service.

no reaction when pushing the button.

i tried several other pins and another raspberry to ensure, nothing is broken.

something very odd:

when narrowing down the problem, i tried (after fresh installation) the following configuration:

[PlayPause]
enabled: True
Type: Button
Pin: 37
pull_up_down: pull_up
functionCall: functionCallPlayerPause

after reloading the service playback stutters, like the button is pushed autmatical (althogh no button connected to the pi). furthermore this configuration destroys the network connectivity of the pi. it looses its wifi ip-adress, connection can only be done with hdmi connected and a keyboard directly attached. this is also reproducable with two different raspis.

so gpio-control seems to be broken in the current developer-version.

any help is appreciated :-)

**Comment by derschelli** [comment]
Just tested with 2.4 master:
same results
gpio buttons not working.

using pin 37 destroys networking:

pi@lina-testbox:~ $ nano ~/RPi-Jukebox-RFID/settings/gpio_settings.ini

[PlayPause]
enabled: True
Type: Button
Pin: 37
pull_up_down: pull_up
functionCall: functionCallPlayerPause


pi@lina-testbox:~ $ sudo systemctl restart phoniebox-gpio-control
pi@lina-testbox:~ $ Network error: Software caused connection abort

disabling pin 37 again repairs networking:
pi@lina-testbox:~ $ nano ~/RPi-Jukebox-RFID/settings/gpio_settings.ini

[PlayPause]
enabled: False
Type: Button
Pin: 29
pull_up_down: pull_up
functionCall: functionCallPlayerPause


pi@lina-testbox:~ $ sudo systemctl restart phoniebox-gpio-control
pi@lina-testbox:~ $ sudo reboot

**Reply by AlvinSchiller** [reply]
Maybe the same problem as https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1964?

Are you still encounter this problem with the new release?

i recently tried some pins and they were working fine. Do you connect them directly to the Pi pins or do you have something in between, like a splitting board or something like that?

**Reply by s-martin** [reply]
@derschelli If you are still experiencing problems, please continue the discussion in #1964, so we keep everything in one place. Thanks

**Reply by derschelli** [reply]
Thank you for your answers. I finished the box without buttons because they didn't work. You need to swipe a card for every function. Quite annoying, so the second box is almost never in use. Maybe i will try to add buttons when I have some spare time. I'll have a look at #1964 then.

**Comment by AlvinSchiller** [comment]
Hi i think you have the same problem with the pin numbering as in the latest report here https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1964#issuecomment-1917989032

Please use the GPIO pin numbers and not the Board pin numbers and try again. See here for further details [Using-GPIO-hardware-buttons#pin-numbering-on-the-rpi](https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Using-GPIO-hardware-buttons#pin-numbering-on-the-rpi) (Board Pin 29 = GPIO Pin 5, Board Pin 37 = GPIO Pin 26)
